Tests/BigIntTests/BigIntTests.swift

```diff
@@ -12,19 +12,44 @@
 import BigInt
 import XCTest
 
-func fac(_ n: BigInt) -> BigInt {
-  var result: BigInt = 1
-  var count = n
-  while count >= 1 {
-    result *= count
-    count -= 1
+extension BigInt {
+
+  static func fac(_ n: BigInt) -> BigInt {
+    precondition(n >= 0, "Factorial of a negative integer is undefined")
+    return stride(from: n, to: 1, by: -1).reduce(into: 1, { $0 *= $1 })
   }
 
-  return result
+  // inspired by https://eli.thegreenplace.net/2009/03/21/efficient-integer-exponentiation-algorithms
+  static func pow(_ lhs: BigInt, _ rhs: BigInt) -> BigInt {
+    let bits_of_n = {
+      (n: BigInt) -> [Int] in
+      var bits: [Int] = []
+      var n = n
+      while n != 0 {
+        bits.append(Int(n % 2))
+        n /= 2
+      }
+
+      return bits
+    }
+
+    var r: BigInt = 1
+    for bit in bits_of_n(rhs).reversed() {
+      r *= r
+      if bit == 1 {
+        r *= lhs
+      }
+    }
+
+    return r
+  }
 }
 
 final class BigIntTests: XCTestCase {
 
+  /// Float64 or Float80 (or Float128)
+  typealias FloatXX = CLongDouble
```

Comment:
Hey so just tested it and it turns out `CLongDouble` is *not* defined on arm linux, either 32 or 64 bit variants.

Reply:
Fixed. I've used the same `#if` condition as found in ComplexTests and RealTests.